src/auth/token-generator.ts

```diff
@@ -254,13 +255,38 @@ export class FirebaseTokenGenerator {
    *                           service account key and containing the provided payload.
    */
   public createCustomToken(uid: string, developerClaims?: {[key: string]: any}): Promise<string> {
+    return this.createCustomTokenInternal(uid, null, developerClaims);
+  }
+
+  /**
+   * Creates a new Firebase Auth Custom token.
+   *
+   * @param {string} uid The user ID to use for the generated Firebase Auth Custom token.
+   * @param {string} tenantId The tenant ID to use for the generated Firebase Auth Custom token.
+   * @param {object} [developerClaims] Optional developer claims to include in the generated Firebase
+   *                 Auth Custom token.
+   * @return {Promise<string>} A Promise fulfilled with a Firebase Auth Custom token signed with a
+   *                           service account key and containing the provided payload.
+   */
+  public createCustomTokenWithTenantId(
+      uid: string, tenantId: string, developerClaims?: {[key: string]: any}): Promise<string> {
+    if (!validator.isNonEmptyString(tenantId)) {
+      throw new FirebaseAuthError(
+        AuthClientErrorCode.INVALID_ARGUMENT,
+        '`tenantId` argument must be a non-empty string uid.');
+    }
+    return this.createCustomTokenInternal(uid, tenantId, developerClaims);
+  }
+
+  private createCustomTokenInternal(
```

Comment:
Alternatively to having two separate createCustomToken*() methods, we could have one with an options object, i.e.:
```javascript
interface CreateCustomTokenOptions {
  tenantId?: string;
  developerClaims?: {[key: string]: any};
}
public createCustomToken(uid: string, options?: CreateCustomTokenOptions) { ... }
```